### ISSUE TITLE
[Slint Preview ver. 1.0.1 Only] [MacOS] [Visual Studio Code] Preview window does not work

### ISSUE THREAD

**Issue by volodg** [slint-ui/slint#2609]
Given a slint layout
When preview layout in Visual Studio Code
Then preview appears in menu bar but does not show up

**Comment by ogoffart** [comment]
Thanks for reporting a bug.

Would you be able to share the code that is no longer working so we can investigate?

**Comment by volodg** [comment]
Yes, you can test layouts from this file for example - https://github.com/biotech-ng/exchange/blob/main/slint_lib/ui/app.slint

**Comment by ogoffart** [comment]
Everything seems to work alright for me.
What component exactly is not working? What are the step i can use to reproduce?  Could you also perhaps send screenshot, that may help me to understand

**Comment by volodg** [comment]
MacOS ver 12.6.3
Issue:
https://user-images.githubusercontent.com/32120/233590225-2593dbff-677a-4a6a-84b2-e4308934ab97.mov

Visual studio Code ver 1.77.3

Slint preview plugin version 1.0.1
<img width="547" alt="Screenshot 2023-04-21 at 10 47 13" src="https://user-images.githubusercontent.com/32120/233590621-d119a3c8-88a0-4c84-8181-c70e152859f0.png">

Version 1.0.0 was working fine

**Comment by ogoffart** [comment]
Thanks for the video. That shows the problem clearly;
But I can't reproduce this problem on Linux. Maybe it's mac specific?
@tronical  could you have a look?   (If i have to guess what break it, i'd say it may be releated to the always-on-top flag)

**Comment by tronical** [comment]
I can't reproduce this either on macOS :-(. @volodg  does it help if you restart vs code?

**Comment by CarbonPool** [comment]
I also confirm, I have this problem on windows

**Comment by ogoffart** [comment]
Something that could explain the behavior is if the preview thread is stuck in an infinite loop.  Is the slint-lsp process taking a lot of CPU in the background?

**Comment by volodg** [comment]
@ogoffart No, id does not

<img width="1120" alt="Screenshot 2023-04-24 at 21 48 01" src="https://user-images.githubusercontent.com/32120/234100599-9e01890c-80b1-4e7e-a493-6ca44e1bf92a.png">

**Comment by tronical** [comment]
@volodg what version of macOS are you running?

**Comment by tronical** [comment]
> Yes, you can test layouts from this file for example - https://github.com/biotech-ng/exchange/blob/main/slint_lib/ui/app.slint

Just to double check: a new vs code instance, open this file, preview the last component, and then nothing shows but you see an entry on the task bar?

**Comment by tronical** [comment]
@CarbonPool do you have some steps that reproduce this?

I have the feeling that this might be content related.

**Comment by volodg** [comment]
@tronical 
MacOS ver.: 12.6.3

Yes, exactly: "nothing shows but you see an entry on the task bar"

**Comment by volodg** [comment]
I don't think it's content related, seems I can not preview any component

**Comment by tronical** [comment]
Do you see anything in the output tab for "Slint LSP"? There should be some messages there. I wonder what it says there when you click "Show Preview".

**Comment by CarbonPool** [comment]
> @CarbonPool do you have some steps that reproduce this?
> 
> I have the feeling that this might be content related.

1. update slint and slint-build to 1.0.1
2. restart vscode or extension

**Comment by volodg** [comment]
@tronical Not sure, where I can find this tab? Maybe this?
<img width="173" alt="Screenshot 2023-04-25 at 09 46 14" src="https://user-images.githubusercontent.com/32120/234209440-d2fc9935-6499-4541-9027-7c1e57fa9e5d.png">

**Comment by ogoffart** [comment]
We mean the output tab: 
![image](https://user-images.githubusercontent.com/959326/234212235-7a97e865-5911-452c-8cf3-e61bccd8d338.png)
If there is no "Slint LSP" there, it means there was no output for the LSP.
But if the LSP crashes, the panic message should be printed there.

**Comment by volodg** [comment]
Nothing special there:

<img width="1357" alt="Screenshot 2023-04-25 at 10 02 18" src="https://user-images.githubusercontent.com/32120/234213403-9b5bdf56-99e8-45e1-bea8-a7ca799735cf.png">

**Comment by ogoffart** [comment]
I actually can reproduce.  I confirm that the version 1.0.1 doesn't work.  What's strange is that the nightly version works.

**Comment by ogoffart** [comment]
What's really strange is that i tried to install previous nightly build, and none of the nightly build between april 19th and april 21st reproduce the problem, despite it should be exactly the same as the 1.0.1 release

**Comment by ogoffart** [comment]
I was able to bisect and found out this was caused by commit abdbb8d8c648f5a8568fd6be5d70019013637b71 just before the release, and this was fixed just after the release in 21a06d1765c0f0558d31a4382250a684bede4fd1

**Comment by ogoffart** [comment]
We will release a version soon. I'd like to thanks everyone involved for reporting the issue and the patience helping with the debugging. (also in #2616 )
